kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/util/NetworkUtil.java

```diff
@@ -125,73 +108,141 @@ public static int convertIp4Address(String ipAddress) {
+                retVal[k + 1] = (byte) (value & 0xFF);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("fullFormIP6Address is invalid: " + fullFormIP6Address, e);
+            }
         }
 
         return retVal;
     }
 
     public static String convertIP6Address(byte[] bytes) {
-        StringBuffer sb = new StringBuffer();
-        for (int i = 0; i < 16; i = i + 2) {
-            sb.append(Integer.toHexString(0xFF & bytes[i]));
-            sb.append(Integer.toHexString(0xFF & bytes[i + 1]));
-            if (i != 14) {
-                sb.append(":");
-            }
+        if ((bytes == null) || (bytes.length != 16)) {
+            throw new IllegalArgumentException("bytes is null or invalid");
         }
-        return sb.toString();
+
+        String[] items = new String[8];
+        for (int i = 0; i < 8; i++) {
+            int k = i * 2;
+            int value = (bytes[k] << 8) & 0xFF00;
+            value |= bytes[k + 1] & 0xFF;
+            items[i] = Integer.toHexString(value);
+        }
+
+        return joinStrings(":", items);
     }
 
     public static String macToString(byte[] mac) {
-        StringBuilder sb = new StringBuilder();
+        if ((mac == null) || (mac.length != 6)) {
+            throw new IllegalArgumentException("mac is null or invalid");
+        }
+
+        String[] items = new String[6];
         for (int i = 0; i < 6; i++) {
             String octet = Integer.toHexString(mac[i] & 0xFF).toUpperCase();
             if (octet.length() == 1) {
-                sb.append("0");
-            }
-            sb.append(octet);
-            if (i != 5) {
-                sb.append(":");
+                octet = "0" + octet;
             }
+
+            items[i] = octet;
         }
-        return sb.toString();
+
+        return joinStrings(":", items);
     }
 
     public static byte[] macToBytes(String mac) {
-        StringTokenizer st = new StringTokenizer(mac, ":");
+        if (mac == null) {
+            throw new IllegalArgumentException("mac is null");
+        }
+
+        String[] items = mac.split("\\:");
+
+        if (items.length != 6) {
+            throw new IllegalArgumentException("mac is invalid: " + mac);
+        }
+
         byte[] bytes = new byte[6];
         for (int i = 0; i < 6; i++) {
-            bytes[i] = (byte) Integer.parseInt(st.nextToken(), 16);
+            String item = items[i];
+            if (item.isEmpty() || (item.length() > 2)) {
+                throw new IllegalArgumentException("mac is invalid: " + mac);
+            }
+
+            try {
+                bytes[i] = (byte) Integer.parseInt(items[i], 16);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("mac is invalid: " + mac, e);
+            }
         }
 
         return bytes;
     }
+
+    private static String joinStrings(String separator, String[] items) {
```

Comment:
I did use Java 8, that is why the original build failed :)

@ctron I have seen a pull request (#775) that mentioned that Kura is switching to Java 8. Is it now allowed to use Java 8?